config/jobs/kubernetes-sigs/cluster-api-provider-azure/cluster-api-provider-azure-presubmits-main.yaml

```diff
@@ -80,6 +80,7 @@ presubmits:
       preset-azure-cred-only: "true"
       preset-azure-anonymous-pull: "true"
       preset-azure-capz-sa-cred: "true"
+      preset-azure-cred-wi: "true"
```

Comment:
Maybe we should create a new optional job, so we can remove the `preset-azure-cred-only: "true"` and `preset-azure-capz-sa-cred: "true"` and not break existing jobs but test that we actually removed everything that we needed to

I am not sure what `preset-azure-capz-sa-cred: "true"` is for. @jackfrancis do you know?

Reply:
Checking if we need access to `AZURE_STORAGE_ACCOUNT` and `AZURE_STORAGE_KEY` in `pull-cluster-api-provider-azure-e2e`

Reply:
We should be able to assign the properly RBAC to the ClientID if it is required.

Reply:
> We should be able to assign the properly RBAC to the ClientID if it is required.

Not sure I understand this. 
Do right RBACs to to Azure Workload ID will solve the issue if the tests start failing on missing `AZURE_STORAGE_ACCOUNT` ?